### PR TITLE
just added some useful features

### PR DESCRIPTION
**Pull request by leabdalla**

### CODE REVIEW

File: instagramcrawler.py [hunk 1]
```diff
@@ -323,12 +327,14 @@ def main():
                         help='Number of posts to download: integer')
     parser.add_argument('-c', '--caption', action='store_true',
                         help='Add this flag to download caption when downloading photos')
+    parser.add_argument('-l', '--headless', type=int, default=0,
```
Comment (JakeCowton) [inline instagramcrawler.py]: I'd recommend storing this in the same way the captions are stored using `action='store_true'`, that way you can just pass `-l` without having to supply `1`
```
parser.add_argument('-l', '--headless', action='store_true',
```

File: instagramcrawler.py [hunk 2]
```diff
@@ -59,8 +59,12 @@ class InstagramCrawler(object):
     """
         Crawler class
     """
-    def __init__(self):
-        self._driver = webdriver.Firefox()
+    def __init__(self, headless=0):
```
Comment (JakeCowton) [inline instagramcrawler.py]: Change this to
`def __init__(self, headless=False):`

File: instagramcrawler.py [hunk 3]
```diff
@@ -59,8 +59,12 @@ class InstagramCrawler(object):
...
+        if(headless == 1):
```
Comment (JakeCowton) [inline instagramcrawler.py]: Change this to `if headless:`

**Review by JakeCowton — CHANGES_REQUESTED** [review]
I've suggested a change to how the headless parameter is passed around the code. It means you don't need to put `-l 1` in the command line and can just use `-l`.

(I made a PR for this on your branch)

**Comment by tzuhsial** [comment]
@JakeCowton @leabdalla 
Thanks!
I'll look at it later
I haven't really written any tests yet, so just ignore the failed check

**Comment by tzuhsial** [comment]
I'll merge it after adding some tests...

**Comment by tzuhsial** [comment]
@leabdalla 
merged

**Pull request merged**